internal/auth/oauth2.go

```diff
@@ -420,13 +426,23 @@ func (as *AuthenticationService) VerifyClientCredentials(ctx context.Context, cl
 	return nil
 }
 
+// sessionCrossOriginSafe sets parameters on the session that allow its use in cross-origin requests.
+// Options are not saved to the database so this must be called whenever a session cookie will be returned to a client.
+//
+// Note browsers require cookies with the same-site policy as 'none' to additionally have the secure flag set.
+func sessionCrossOriginSafe(session *sessions.Session, secure bool) *sessions.Session {
+	session.Options.Secure = secure                  // Ensures only sent with HTTPS
+	session.Options.HttpOnly = false                 // Allow Javascript to read it
```

Comment:
No they call whoami to get details

Reply:
Good point @SimoneDutto I went and tried my test again, setting HttpOnly = true and the browser still sends the cookie. Had a google and I see from [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies)
>A cookie with the HttpOnly attribute can't be modified by JavaScript, for example using [Document.cookie](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie); it can only be modified when it reaches the server. Cookies that persist user sessions for example should have the HttpOnly attribute set — it would be really insecure to make them available to JavaScript. This precaution helps mitigate cross-site scripting ([XSS](https://developer.mozilla.org/en-US/docs/Web/Security/Types_of_attacks#cross-site_scripting_xss)) attacks.

So I'll change this to true.

Reply:
yes, this is more secure! thank you for this